utils/config.go

```diff
@@ -689,3 +709,75 @@ func IsLeader() bool {
 		return true
 	}
 }
+
+func ValidateAmazonS3Endpoint(cfg *model.Config) {
+	_, ok := awsS3EndpointMap[*cfg.FileSettings.AmazonS3Endpoint]
+	if !ok {
+		*cfg.FileSettings.AmazonS3Endpoint = model.FILE_SETTINGS_DEFAULT_AMAZON_S3_ENDPOINT
+		l4g.Warn(T("utils.config.set_amazon_endpoint"), model.FILE_SETTINGS_DEFAULT_AMAZON_S3_ENDPOINT)
+	}
+}
+
+func ValidateAmazonS3Region(cfg *model.Config) {
+	valid := false
+	for _, region := range awsS3EndpointMap {
+		if region == *cfg.FileSettings.AmazonS3Region {
+			valid = true
+			break
+		}
+	}
+
+	if !valid {
+		*cfg.FileSettings.AmazonS3Region = model.FILE_SETTINGS_DEFAULT_AMAZON_S3_REGION
+		l4g.Warn(T("utils.config.set_amazon_region"), model.FILE_SETTINGS_DEFAULT_AMAZON_S3_REGION)
+	}
+}
+
+func ValidateAmazonS3Bucket(cfg *model.Config) *model.AppError {
+	if *cfg.FileSettings.AmazonS3Bucket == "" {
+		return model.NewAppError("ValidateAmazonS3Bucket", "utils.config.bucket_empty.app_error", nil, "", http.StatusBadRequest)
+	}
+
+	endpoint := *cfg.FileSettings.AmazonS3Endpoint
+	bucket := *cfg.FileSettings.AmazonS3Bucket
+	accessKey := cfg.FileSettings.AmazonS3AccessKeyId
+	secretKey := cfg.FileSettings.AmazonS3SecretAccessKey
+	secure := *cfg.FileSettings.AmazonS3SSL
+
+	s3Clnt, err := s3.New(endpoint, accessKey, secretKey, secure)
+	if err != nil {
+		return model.NewAppError("ValidateAmazonS3Bucket", "utils.config.bad_connection_to_s3_or_minio.app_error", nil, err.Error(), http.StatusBadRequest)
+	}
+
+	bucketLocation, err := s3Clnt.GetBucketLocation(bucket)
```

Comment:
Does s3manager's [GetBucketRegion](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/s3manager/#GetBucketRegion) work with Minio? If so, we should use it instead since it doesn't require additional IAM permissions to use.

Reply:
Not sure, but might work. I looked at couple of issues in minio repo and it seemed working. Only that we are not yet using `aws-sdk-go` in our repo at the moment.

Reply:
Yeah... I think using `aws-sdk-go` would allow us to get rid of some of this code and make things easier, but it is a massive dependency. It's one of the only Go packages I'm hesitant to import due to its size because it does have a noticeable impact on build time. So... /shrug